wsd/DocumentBroker.cpp

```diff
@@ -3085,7 +3097,10 @@ bool DocumentBroker::sendUnoSave(const std::shared_ptr<ClientSession>& session,
     // Note: It's odd to capture these here, but this function is used from ClientSession too.
     const bool autosave = isAutosave || (_unitWsd && _unitWsd->isAutosave());
     const bool backgroundConfigured = (autosave && _backgroundAutoSave) || _backgroundManualSave;
-    const bool background = forceBackgroundEnv || (!finalWrite && backgroundConfigured);
+    const bool canBackground = forceBackgroundEnv || (!finalWrite && backgroundConfigured);
+    constexpr std::size_t MaxFailureCountForBackgroundSaving = 7;
```

Comment:
I -hope- this is really rare; I don't like the hard-coded 7 - if we fail to background save - lets just turn off background save for this document - and log the error.

Reply:
I didn't want to disable the good performance on the first unlucky timeout or failure (because we do adjust the timeout after the first save result). Of course this is arbitrary. It can be any number, including 1.

Reply:
Now giving only 1 extra chance (i.e. we tolerate background saving failures only twice, at most), before disabling it altogether (for the given document).